### PR TITLE
[CI] Add CPU C++ test stage

### PR DESCRIPTION
**Pull request by larroy**
## Description ##
We were not running C++ test compiled without GPU and without MKL, which had test failures, such as https://github.com/apache/incubator-mxnet/pull/13409

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: src/operator/nn/activation-inl.h [hunk 1]
```diff
@@ -199,13 +199,8 @@ void ActivationGradCompute(const nnvm::NodeAttrs& attrs,
                            const std::vector<OpReqType>& req,
                            const std::vector<TBlob>& outputs) {
   const ActivationParam& param = nnvm::get<ActivationParam>(attrs.parsed);
-#if (MXNET_USE_CUDNN == 1 || MXNET_USE_MKLDNN == 1)
-  bool relu = param.act_type == activation::kReLU;
+  const bool relu = param.act_type == activation::kReLU;
   CHECK_EQ(inputs.size(), relu ? 2U : 3U);
```
Comment (lebeg) [inline src/operator/nn/activation-inl.h]: This code was added in https://github.com/apache/incubator-mxnet/pull/11827 . Maybe @samskalicky can help here and say whether it's needed

**Comment by stu1130** [comment]
@mxnet-label-bot add [pr-awaiting-review]
Thanks for your contribution! @larroy

**Comment by marcoabreu** [comment]
Please wait with merging this PR until the Jenkinsfile refactoring has been completed

**Comment by vandanavk** [comment]
@mxnet-label-bot update [pr-awaiting-testing]

File: src/operator/nn/activation-inl.h (hunk 1 above)
Comment (larroy) [reply src/operator/nn/activation-inl.h]: I fixed all this stuff in https://github.com/apache/incubator-mxnet/pull/13409  just keeping the CI pipeline here until @marcoabreu  green lights.

**Comment by roywei** [comment]
@larroy ping, is this still needed? thanks!

**Comment by larroy** [comment]
@roywei is needed.  @marcoabreu  pls review.

**Comment by larroy** [comment]
Ready to merge @marcoabreu

**Comment by larroy** [comment]
@lebeg no, I will remove them.

**Pull request merged**